sdk/lib/rtl/rangelist.c

```diff
@@ -13,16 +13,478 @@
 #define NDEBUG
 #include <debug.h>
 
+/* GLOBALS ******************************************************************/
+
+//extern PAGED_LOOKASIDE_LIST RtlpRangeListEntryLookasideList;
+
 /* TYPES ********************************************************************/
 
-typedef struct _RTL_RANGE_ENTRY
-{
-    LIST_ENTRY Entry;
-    RTL_RANGE Range;
-} RTL_RANGE_ENTRY, *PRTL_RANGE_ENTRY;
+/* RTLP_RANGE_LIST_ENTRY == RTL_RANGE + ListEntry */
+
+/*  FIXME ? ntddk.h */
+/* Flags for public functions: RtlFindRange(), RtlIsRangeAvailable() */
+#define RTL_RANGE_LIST_SHARED_OK           1
+#define RTL_RANGE_LIST_NULL_CONFLICT_OK    2
+/* ? ntddk.h */
+
+/* RTLP_RANGE_LIST_ENTRY.PrivateFlags */
+#define RTLP_ENTRY_IS_MERGED  1
+
 
 /* FUNCTIONS ***************************************************************/
 
+static
+PRTLP_RANGE_LIST_ENTRY
+FASTCALL
+RtlpEntryFromLink(PVOID Link)
+{
+    return CONTAINING_RECORD(Link, RTLP_RANGE_LIST_ENTRY, ListEntry);
+}
+
+static
+BOOLEAN
+FASTCALL
+IsRangesIntersection(
+    _In_ PRTLP_RANGE_LIST_ENTRY Entry1,
+    _In_ PRTLP_RANGE_LIST_ENTRY Entry2)
+{
+    /* Not intersection when:
+       (Entry1->Start - Entry1->End) ... (Entry2->Start -
+       (Entry2->Start - Entry2->End) ... (Entry1->Start -
+    */
+
+    if (((Entry2->Start > Entry1->Start && Entry2->Start > Entry1->End) ||
+         (Entry1->Start > Entry2->Start && Entry1->Start > Entry2->End)))
+    {
+        /* No intersection */
+        return FALSE;
+    }
+    else
+    {
+        /* Entry1 and Entry2 overlap */
+        return TRUE;
+    }
```

Comment:
The code can be simplified:
```suggestion
    return (Entry1->Start < Entry2->End && Entry2->Start < Entry1->End);
```

Reply:
Your variant (as it stands) may return an incorrect value. It is, of course, shorter, but more difficult to understand. If optimization priority is more important, then your option is preferable. If the priority is more understandable (for a human :)) code, then it is better to leave it as it is. For me personally, it doesn't really matter. For me, compatibility and functionality remains the highest priority. So I can easily change if you prefer your own version.